### DISCUSSION TITLE
Milvus Backup Questions for Standalone Docker Compose: Is ./milvus-backup create -n <backup_name> better than just copying files? What are the benefits?

### DISCUSSION THREAD

**Discussion by martinenkoEduard** [milvus-io/milvus#34568] [Q&A and General discussion]
Milvus Backup Questions for Standalone Docker Compose: Is ./milvus-backup create -n <backup_name> better than just copying files? What are the benefits?

What are the benefits of executing ./milvus-backup create -n <backup_name>? Am I correct in assuming that I can perform this on the working instance of Milvus?

Where should I enter this command? Do I need to enter this command using Docker: 
docker exec -it <mycontainer> bash?
Does it copies files inside MINIO or does it create a separate file?

**Comment by yhmo** [comment]
milvus-backup tool requires the milvus + minio.
There are two approaches to deploy milvus standalone:
1. https://milvus.io/docs/install_standalone-docker-compose.md   three containers (milvus + minio + etcd)
2. https://milvus.io/docs/install_standalone-docker.md    embed milvus, only one container, this milvus uses the local disk to replace minio

If your milvus is embed milvus, the milvus-backup cannot work.
If your milvus is milvus + minio + etcd, the command `./milvus-backup create -n <backup_name>` is to backup all the collection of this milvus, the data will be uploaded to the path /[bucket name of the milvus]/[buckup_name]. Then you can copy the backup data to another minio of another milvus. Then use `./milvus-backup restore <backup_name>` to restore the data into the milvus.


No need to use docker exec to enter the container.

**Comment by yhmo** [comment]
A doc about how to use milvus-backup to backup/restore in one milvus or between multi milvus:
[milvus-backup tool practice.pdf](https://github.com/user-attachments/files/16158172/milvus-backup.tool.practice.pdf)

**Comment by martinenkoEduard** [comment]
Is
https://milvus.io/docs/install_standalone-docker.md embed milvus, only one container, this milvus uses the local disk to replace minio
faster that version that uses minio?

**Reply by martinenkoEduard** [reply]
Like if it is "embed milvus, only one container, this milvus uses the local disk to replace minio" it sound like it should be much faster? Is it the case?

**Reply by xiaofan-luan** [reply]
dosn't really true.

Most milvus index will load data into memory before search. So only load performance will be affected

**Comment by wayblink** [comment]
> Milvus Backup Questions for Standalone Docker Compose: Is ./milvus-backup create -n <backup_name> better than just copying files? What are the benefits?
> 
> What are the benefits of executing ./milvus-backup create -n <backup_name>? Am I correct in assuming that I can perform this on the working instance of Milvus?
> 
> Where should I enter this command? Do I need to enter this command using Docker: docker exec -it bash? Does it copies files inside MINIO or does it create a separate file?

milvus-backup not only backup files but also essential metadata. We can't restore a milvus cluster only with copied files
